### PR TITLE
Fix td avro parsing

### PR DESCRIPTION
**Pull request by umanwizard**
See the [relevant section of the spec](https://avro.apache.org/docs/current/spec.html#json_encoding).

> The value of a union is encoded in JSON as follows:
>
> * if its type is null, then it is encoded as a JSON null;
> * otherwise it is encoded as a JSON object with one name/value pair whose name is the type's name and whose value is the recursively encoded value. For Avro's named types (record, fixed or enum) the user-specified name is used, for other types the type name is used.

---
This change is [Reviewable](https://reviewable.io/reviews/materializeinc/materialize/4266)

### CODE REVIEW

File: src/testdrive/src/format/avro.rs [hunk 1]
```diff
@@ -137,27 +138,62 @@ pub fn from_json(json: &JsonValue, schema: SchemaNode) -> Result<Value, String>
             }
             Ok(builder.avro())
         }
-        (val, SchemaPiece::Union(us)) => {
+        (val, /*JsonValue::Object(items),*/ SchemaPiece::Union(us)) => {
```
Comment (benesch) [inline src/testdrive/src/format/avro.rs:141]: ?

File: src/testdrive/src/format/avro.rs [hunk 2]
```diff
@@ -137,27 +138,62 @@ pub fn from_json(json: &JsonValue, schema: SchemaNode) -> Result<Value, String>
...
+                        n_variants: variants.len(),
+                        null_variant,
+                    })
+                } else {
+                    Err("No `null` value in union schema.".to_string())
+                };
+            }
+            let items = match val {
+                JsonValue::Object(items) => items,
+                _ => return Err(format!("Union scheme element must be `null` or a map from type name to value; found {:?}", val))
```
Comment (benesch) [inline src/testdrive/src/format/avro.rs:160]: nit: s/scheme/schema/

**Review by benesch — APPROVED** [review]

File: src/testdrive/src/format/avro.rs (hunk 1 above)
Comment (umanwizard) [reply src/testdrive/src/format/avro.rs:141]: good catch

File: src/testdrive/src/format/avro.rs (hunk 2 above)
Comment (umanwizard) [reply src/testdrive/src/format/avro.rs:160]: idem

**Pull request merged**